src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/ThreadStackTrace.java

```diff
@@ -46,9 +46,6 @@ public StackFrameInfo stackFrameAt(int index) {
     }
 
     public void dumpStack(int maxDepth) {
-        if (!thread.isJavaThread()) {
-            return;
-        }
```

Comment:
Q: I wonder why does `!thread.isHiddenFromExternalView()` not used here. Have we always dumped the hidden from external view threads?

Reply:
It's only used by the code below, which has already done the check:
```
           if (threadObj != null && !thread.isExiting() && !thread.isHiddenFromExternalView()) {

                // dump thread stack trace
                ThreadStackTrace st = new ThreadStackTrace(thread);
                st.dumpStack(-1);
                numThreads++;
```

Reply:
I see it now. Thanks!